addons/hr_holidays/tests/test_allocations.py

```diff
@@ -60,6 +61,29 @@ def setUpClass(cls):
                 ]
         })
 
+        cls.calendar_0h = cls.env['resource.calendar'].create({
+            'name': 'Calendar - 0H',
+            'company_id': cls.company.id,
+            'attendance_ids': [Command.clear()],
+        })
+
+        cls.employee_0h = cls.env['hr.employee'].create({
+            'name': 'My Employee 0 Hours',
+            'company_id': cls.company.id,
+            'department_id': cls.department.id,
+            'category_ids': [Command.link(cls.category_tag.id)],
+            'resource_calendar_id': cls.calendar_0h.id,
+        })
```

Comment:
This shouldn't be defined here, setup records are meant to be shared and re-used by multiple test methods, in this case you only use it in your method, create it there. Same for the 2 other records.

But actually, for this one, you don't even need to create this one, just use the existing `cls.employee` in your test method. Just write this at the top of your method:
```py
cls.employee.resource_calendar_id = cls.calendar_0h
```